src/Parser/CachedParser.php

```diff
@@ -30,66 +26,35 @@ public function __construct(
 	 */
 	public function parseFile(string $file): array
 	{
-		if ($this->cachedNodesByStringCountMax !== 0 && $this->cachedNodesByStringCount >= $this->cachedNodesByStringCountMax) {
-			$this->cachedNodesByString = array_slice(
-				$this->cachedNodesByString,
+		if ($this->cachedNodesByFileCount !== 0 && $this->cachedNodesByFileCount >= $this->cachedNodesByStringCountMax) {
+			$this->cachedNodesByFile = array_slice(
+				$this->cachedNodesByFile,
 				1,
 				preserve_keys: true,
 			);
 
-			--$this->cachedNodesByStringCount;
+			--$this->cachedNodesByFileCount;
 		}
 
-		$sourceCode = FileReader::read($file);
-		if (!isset($this->cachedNodesByString[$sourceCode]) || isset($this->parsedByString[$sourceCode])) {
-			$this->cachedNodesByString[$sourceCode] = $this->originalParser->parseFile($file);
-			$this->cachedNodesByStringCount++;
-			unset($this->parsedByString[$sourceCode]);
+		if (!isset($this->cachedNodesByFile[$file])) {
+			$this->cachedNodesByFile[$file] = $this->originalParser->parseFile($file);
+			$this->cachedNodesByFileCount++;
 		}
 
-		return $this->cachedNodesByString[$sourceCode];
+		return $this->cachedNodesByFile[$file];
 	}
 
 	/**
 	 * @return Node\Stmt[]
 	 */
 	public function parseString(string $sourceCode): array
```

Comment:
hmm I wonder whether/why we need this `parseString` function. it seems its only used from tests and everywhere else we read a file to pass the contents into `parseString`.

I think we could remove some unnecessary file-read operations by dropping `parseString` and just use `parseFile` everywhere with PHPStan 3.x ... wdyt?